front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -1,63 +1,43 @@
 import React, { useState, useEffect } from 'react';
 
-import { DynamicDataSheetGrid } from 'react-datasheet-grid';
+import cx from 'classnames';
+import { DynamicDataSheetGrid, type DataSheetGridProps } from 'react-datasheet-grid';
 import { useTranslation } from 'react-i18next';
 
-import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
 import { useOsrdConfActions } from 'common/osrdContext';
 import { isVia } from 'modules/pathfinding/utils';
 import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
 import type { PathStep } from 'reducers/osrdconf/types';
 import { useAppDispatch } from 'store';
-import { removeElementAtIndex } from 'utils/array';
-import { marginRegExValidation } from 'utils/physics';
 
-import { useInputColumns } from './TimeStopsColumns';
-import type { PathWaypointColumn } from './types';
-import { formatSuggestedViasToRowVias } from './utils';
+import { marginRegExValidation } from './consts';
+import { formatSuggestedViasToRowVias } from './helpers/utils';
+import { useTimeStopsColumns } from './TimeStopsColumns';
+import { TableType } from './types';
+import type { PathWaypointRow } from './types';
 
-const WITH_KP = true;
+export const WITH_KP = true;
 
 type TimesStopsProps = {
-  pathProperties: ManageTrainSchedulePathProperties | undefined;
+  allWaypoints?: SuggestedOP[];
   pathSteps?: PathStep[];
   startTime?: string;
+  tableType: TableType;
+  cellClassName?: DataSheetGridProps['cellClassName'];
+  stickyRightColumn?: DataSheetGridProps['stickyRightColumn'];
 };
 
-type DeleteButtonProps = {
-  removeVia: () => void;
-  rowIndex: number;
-  rowData: PathWaypointColumn;
-  pathProperties: ManageTrainSchedulePathProperties;
-  pathSteps: PathStep[];
-};
-
-const createDeleteViaButton = ({
-  removeVia,
-  rowIndex,
-  rowData,
-  pathProperties,
-  pathSteps,
-}: DeleteButtonProps) => {
-  const isRowVia =
-    rowIndex !== 0 &&
-    rowIndex !== pathProperties.allWaypoints?.length - 1 &&
-    isVia(pathSteps, rowData, WITH_KP);
-  if (isRowVia) {
-    return (
-      <button type="button" onClick={removeVia}>
-        ❌
-      </button>
-    );
-  }
-  // eslint-disable-next-line react/jsx-no-useless-fragment
-  return <></>;
-};
-
-const TimesStops = ({ pathProperties, pathSteps = [], startTime }: TimesStopsProps) => {
+const TimesStops = ({
+  allWaypoints,
+  pathSteps = [],
+  startTime,
+  tableType,
+  cellClassName,
+  stickyRightColumn,
+}: TimesStopsProps) => {
+  const isInputTable = tableType === TableType.Input;
   const { t } = useTranslation('timesStops');
-
-  if (!pathProperties) {
+  if (!allWaypoints) {
```

Comment:
You should do the return after the useEffect, otherwise I encounter this error when clicking too quickly on the times and stops tab
![image](https://github.com/user-attachments/assets/c1ab373d-b6fe-466b-9f43-4bdb296b5d61)

Reply:
This bug is already on dev and should be fixed in another PR: https://github.com/OpenRailAssociation/osrd/issues/8114 :+1: